examples/morpheusvm/actions/transfer_test.go

```diff
@@ -0,0 +1,128 @@
+	for i := range addrSlice {
+		addrSlice[i] = 1
+	}
+	oneAddr, err := codec.ToAddress(addrSlice)
+	require.NoError(err)
+
+	tests := []chaintest.ActionTest{
+		{
+			Name:  "ZeroTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 0,
+			},
+			ExpectedErr: ErrOutputValueZero,
+		},
+		{
+			Name:  "InvalidStateKey",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State:       ts.NewView(make(state.Keys), map[string][]byte{}),
+			ExpectedErr: tstate.ErrInvalidKeyOrPermission,
+		},
+		{
+			Name:  "NotEnoughBalance",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				keys.Add(string(emptyBalanceKey), state.Read)
+				tsv := ts.NewView(keys, map[string][]byte{})
+				return tsv
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		{
+			Name:  "SelfTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chaintest.NewInMemoryStore()
+				require.NoError(storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				keys.Add(string(emptyBalanceKey), state.All)
+				return ts.NewView(keys, store.Storage)
+			}(),
+			Assertion: func(store state.Mutable) bool {
+				balance, err := storage.GetBalance(context.Background(), store, codec.EmptyAddress)
+				require.NoError(err)
+				return balance == 1
+			},
+		},
+		{
+			Name:  "OverflowBalance",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: math.MaxUint64,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chaintest.NewInMemoryStore()
+				require.NoError(storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				keys.Add(string(emptyBalanceKey), state.All)
+				return ts.NewView(keys, store.Storage)
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		{
+			Name:  "SimpleTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    oneAddr,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chaintest.NewInMemoryStore()
+				require.NoError(storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				keys.Add(string(emptyBalanceKey), state.All)
+				keys.Add(string(storage.BalanceKey(oneAddr)), state.All)
+				return ts.NewView(keys, store.Storage)
+			}(),
+			Assertion: func(store state.Mutable) bool {
+				receiverBalance, err := storage.GetBalance(context.Background(), store, oneAddr)
+				require.NoError(err)
+				senderBalance, err := storage.GetBalance(context.Background(), store, codec.EmptyAddress)
+				require.NoError(err)
+				return receiverBalance == 1 && senderBalance == 0
```

Comment:
+1 to adding a parameter, but I think it should be a `*testing.T` for flexibility. Users can wrap it if they want, but aren't forced to.

> We should also avoid using the local version of `require` within this function.

:100:

Reply:
Thanks for this feedback.
I have renamed the `require` that was used on the global test suite to `req` since it is less used and it is more natural to type `require`. Also removed the `bool` return from the `Assertion` function.

Reply:
Taking @ARR4N suggestion ! Thanks !